sql/analyze_test.go

```diff
@@ -207,11 +207,10 @@ func TestSimplifyExpr(t *testing.T) {
 		{`i LIKE 'foo%'`, `i >= 'foo' AND i < 'fop'`},
 		{`i LIKE 'foo_'`, `i >= 'foo' AND i < 'fop'`},
 		{`i LIKE 'bar_foo%'`, `i >= 'bar' AND i < 'bas'`},
-		// TODO(pmattis): unsupported comparison operator: SIMILAR TO
-		// {`i SIMILAR TO '.*'`, `true`},
-		// {`i SIMILAR TO 'foo'`, `i = 'foo'`},
-		// {`i SIMILAR TO 'foo.*'`, `i >= 'foo' AND i < 'fop'`},
-		// {`i SIMILAR TO '(foo|foobar).*'`, `i >= 'foo' AND i < 'fop'`},
+		{`i SIMILAR TO '.*'`, `true`},
+		{`i SIMILAR TO 'foo'`, `i = 'foo'`},
+		{`i SIMILAR TO 'foo.*'`, `i >= 'foo' AND i < 'fop'`},
+		{`i SIMILAR TO '(foo|foobar).*'`, `i >= 'foo' AND i < 'fop'`},
```

Comment:
I exported `SimilarEscape` from the `parser` package, and used it in `simplifyComparisonExpr` to allow for correct functionality with SQL:2008 regexp patterns.